### PR TITLE
Switch from dotnet/msquic to microsoft/msquic

### PR DESCRIPTION
**Pull request by ManickaP**
In order to archive https://github.com/dotnet/msquic we have to get rid of the references. This change replaces the dotnet repo with the official MsQuic one.

### CODE REVIEW

**Comment by CarnaViire** [comment]
There seem to be build failures @ManickaP

File: src/alpine/3.16/helix/amd64/Dockerfile [hunk 1]
```diff
@@ -69,10 +69,10 @@ RUN apk update && apk add --no-cache && \
         openssl-dev \
         perl && \
     cd /tmp && \
-    git clone --depth 1 --single-branch --branch main --recursive https://github.com/dotnet/msquic && \
-    cd msquic/src/msquic && \
+    git clone --depth 1 --single-branch --branch main --recursive https://github.com/microsoft/msquic && \
```
Comment (wfurt) [inline src/alpine/3.16/helix/amd64/Dockerfile:72]: One option here is to track MsQuic releases. That would match other platforms where we consume published binaries. 
The `dotnet/msquic` give us some control in the past.  
Pointing at `main` may cause some instabilities and surprises but on the other hand it allows us to see issues sonner.

Comment (ManickaP) [reply src/alpine/3.16/helix/amd64/Dockerfile:72]: @CarnaViire suggested the same thing, I'll change it to get this in.

**Comment by jkoritzinsky** [comment]
Related to https://github.com/dotnet/dotnet-buildtools-prereqs-docker/issues/990, we should move these docker images to use multi-stage images that build msquic in one stage and set up the deployed environment (with the built msquic) in another.

**Comment by CarnaViire** [comment]
@jkoritzinsky The problem is that the build scripts (and therefore the produced binaries) are different, depending on the architecture (x64, arm, arm64) and openssl version (1.1, 3.0). So out of the 8 updated images here, 6 would have _different_ MsQuic, so the environment _cannot_ be reused between them.

Note that Alpine is the only platform where we don't have any official MsQuic packages (tracking issue https://github.com/dotnet/runtime/issues/83789), so we have to resort to the manual build here. All the other images install prebuilt binaries from packages.microsoft.com

**Comment by jkoritzinsky** [comment]
I'm not saying we use a "builder" image or something like that to share between the images. I just mean that we do a second FROM statement and install the tools needed for runtime in the second stage (and copy msquic from the first stage) instead of having the build tools and msquic source on the image that runs in Helix.

**Comment by ManickaP** [comment]
> I'm not saying we use a "builder" image or something like that to share between the images. I just mean that we do a second FROM statement and install the tools needed for runtime in the second stage (and copy msquic from the first stage) instead of having the build tools and msquic source on the image that runs in Helix.

I'm not following this line of thought. And is it necessary for this PR or should it rather be resolved separately? After all, I'm just changing from where we take msquic sources, nothing else.

**Comment by jkoritzinsky** [comment]
It's not required for this PR. I can take a look at doing it after this is merged as an example.

**Review by liveans — APPROVED** [review]

**Comment by rzikm** [comment]
Since we opt for consuming the released MsQuic versions, would it make sense to (always) consume the latest published packages where we can? (Ubuntu etc.)

The change does not have to be in this PR, but something to consider for the mentioned future improvements.

**Review by rzikm — APPROVED** [review]

**Comment by CarnaViire** [comment]
> Since we opt for consuming the released MsQuic versions, would it make sense to (always) consume the latest published packages where we can? (Ubuntu etc.)

@rzikm We do already:
https://github.com/dotnet/dotnet-buildtools-prereqs-docker/blob/0b9f68c13626e2e15d916d1565191501b2e0efc5/src/ubuntu/22.04/helix/amd64/Dockerfile#L54-L56
Alpine is the only one where we cannot install the official packages.

**Pull request merged**